### PR TITLE
[WIP] Warn kernel 4.0 

### PR DESCRIPTION
**Pull request by pacoxu**
xref https://github.com/kubernetes/kubernetes/issues/116799 as there are still no conclusions.

### CODE REVIEW

File: validators/kernel_validator_test.go [hunk 1]
```diff
@@ -32,55 +32,69 @@ func TestValidateKernelVersion(t *testing.T) {
...
 			if !test.err {
 				assert.Nil(t, err, "Expect error not to occur with kernel version %q", test.version)
 			} else {
 				assert.NotNil(t, err, "Expect error to occur with kenrel version %q", test.version)
 			}
+			if !test.warn {
+				assert.Nil(t, warn, "Expect warn not to occur with kernel version %q", test.version)
+			} else {
+				assert.NotNil(t, warn, "Expect warn to occur with kenrel version %q", test.version)
+			}
```
Comment (neolit123) [inline validators/kernel_validator_test.go:97]:
```suggestion
			if !test.warn {
				assert.Nil(t, warn, "Expect warning not to occur with kernel version %q", test.version)
			} else {
				assert.NotNil(t, warn, "Expect warning to occur with kernel version %q", test.version)
			}
```

Comment (neolit123) [reply validators/kernel_validator_test.go:97]: there is "kenrel" typo in the old code too.

**Review by neolit123 — COMMENTED** [review]
the Versions / SuggestedVersions separation seems OK.
ideally should not have it and just say "version >= X" is required.
it might be a bit confusing for users what it means to have a "suggested version". they might not know what does not work.

either way, let's wait for comments by sig node on the linked issue.

**Comment by pacoxu** [comment]
> it might be a bit confusing for users what it means to have a "suggested version". they might not know what does not work.

I renamed it from suggested to recommended. And I add RecommendedNote to make the message more readable.

**Comment by pacoxu** [comment]
/close

I may work on this once cgroup v1 is not supported and iptables is not supported oneday. ❓

**Pull request closed without merging**